husc/screens/cellomics.py

```diff
@@ -0,0 +1,366 @@
+    return scale_image
+
+
+def stack_channels(images, order=[0, 1, 2]):
+    """Stack multiple image files to one single, multi-channel image.
+
+    Parameters
+    ----------
+    images : list of array, shape (M, N)
+        The images to be concatenated. List should contain
+        three images. Entries 'None' are considered to be dummy
+        channels
+    order : list of int, optional
+        The order the channels should be in in the final image.
+
+    Returns
+    -------
+    conat_image : array, shape (M, N, 3)
+        The concatenated, three channel image.
+
+    Examples
+    --------
+    >>> image1 = np.ones((2, 2)) * 1
+    >>> image2 = np.ones((2, 2)) * 2
+    >>> joined = stack_channels((image1, image2, None))
+    >>> joined.shape
+    (2, 2, 3)
+    """
+    m = images[0].shape[0]
+    n = images[0].shape[1]
+    concat_image = np.zeros((m, n, 3))
+    for i in range(0, 3):
+        if images[order[i]] is not None:
+            concat_image[:, :, i] = images[order[i]]
+    return concat_image
+
+
+def snail_stitch(fns):
+    """Run right, clockwise spiral/snail stitching of 25 Cellomics TIFs.
+
+    Runs clockwise stitching of the images. The spiral begins in the
+    center of the image, moves to the right and circles around in a clockwise
+    manner.
+
+    Parameters
+    ----------
+    fns : list of array, shape (M, N)
+        The list of 25 image files to be stitched together.
+
+    Returns
+    -------
+    stitched_image : array, shape (5*M, 5*N)
+        The stitched image.
+    """
+    fns.sort()
+    order = [[20, 21, 22, 23, 24],
+             [19, 6, 7, 8, 9],
+             [18, 5, 0, 1, 10],
+             [17, 4, 3, 2, 11],
+             [16, 15, 14, 13, 12]]
+
+    image0 = io.imread(fns[0])
+    m = image0.shape[0]
+    n = image0.shape[1]
+    stitched_image = np.zeros((5*m, 5*n))
+    for i in range(0, 5):
+        for j in range(0, 5):
+            index = order[i][j]
+            image = io.imread(fns[index])
+            stitched_image[m*i:m*(i+1), n*j:n*(j+1)] = image
+    return stitched_image
+
+
+def make_key2file(fns):
+    """Return a dictionary mapping well co-ordinates to filenames.
+
+    Returns a dictionary where key are (plate, well) co-ordinates and
+    values are lists of images corresponding to that plate and well.
+
+    Parameters
+    ----------
+    fns : list of string
+        A list of Cellomics TIF files.
+
+    Returns
+    -------
+    wellchannel2file : dict {tuple : list of string}
+        The dictionary mapping the (plate, well) co-ordinate to
+        a list of files corresponding to that well.
+    """
+    wellchannel2file = coll.defaultdict(list)
+    for fn in fns:
+        fn_base = os.path.basename(fn)
+        file_info = cellomics_semantic_filename(fn_base)
+        coord = (file_info['plate'], file_info['well'])
+        wellchannel2file[coord].append(fn)
+    return wellchannel2file
+
+
+def get_by_ext(path, extension, full=True):
```

Comment:
This seems more generally useful. How about:

- moving it to a more general module, e.g. `pathutils.py`.
- adding support for *any* glob pattern, rather than just extension. e.g. `*.tif`. Use [`fnmatch.filter`](https://docs.python.org/2/library/fnmatch.html#fnmatch.filter) for this.
- renaming it to e.g. `all_matching_files(path, glob, full_path=True)`
- if you intended to recurse subdirectories, make it very clear in the docstring (it's not currently). If not, use `os.listdir` instead of `os.walk`.
- either remove `sort` from the docstring or actually add it. =P